### ISSUE TITLE
Array with nan's shows extra pixels when plotted

### ISSUE THREAD

**Issue by fpavogt** [aplpy/aplpy#357]
Hi everyone,

Stumbled onto something weird: when I plot an array using nan's to hide regions with no data (e.g. in- and outside of the two ellipses in the attached plot), aplpy "creates" extra pixels in the empty regions. 

If instead of nan's I use 0's to fill all the empty regions, the image looks as it should (modulo the blue background which I wanted grey). The only line I change to get this behavior is shown in the code snippet below.

This seems similar to issue #289.  

Any ideas what is going on here ?

```python
plt.close(3)
fig3 = plt.figure(3, figsize=(6.93,7))
fn = os.path.join(prod_loc,'torus_vmap.fits')
hdu = fits.open(fn)

# Uncomment this line to display the correct image (with blue background)
#hdu[0].data[np.isnan(hdu[0].data)] = 0

ax1 = aplpy.FITSFigure(hdu, figure=fig3, subplot=[0.22,0.12,0.7,0.75], north=True)
ax1.show_colorscale(cmap='RdBu_r',interpolation='nearest',vmin = torus_vmean-80, vmax= torus_vmean+80)
ax1.set_nan_color((0.5,0.5,0.5))
```

![torus_vmap_0](https://user-images.githubusercontent.com/13974615/28784064-4ac60f5a-75e0-11e7-96da-8295b986d91e.png)
![torus_vmap_nan](https://user-images.githubusercontent.com/13974615/28784063-4ac51212-75e0-11e7-9eae-54461bd1e134.png)

**Comment by keflavich** [comment]
Agreed that it looks related.  What happens if you set `interpolation='none'` in the `show_colorscale` call?

**Comment by fpavogt** [comment]
Setting ``interpolation = 'none'`` gives the exact same behavior (for both the nan and 0 case).

**Comment by fpavogt** [comment]
I looked deeper into this, and found that Montage might be to blame here. Specifically, the problem arises when I set ``north=True`` in the call to ``FITSFigure``. The array in my Fits file is already oriented North, so in the present case, Montage should not be doing anything. This seems to be the case when the empty regions are filled with 0's, but breaks with nan's. 

For my specific case, I can just set ``north=False`` and I get the plots I need with nan's in the correct locations. To properly fix this issue will require to pass it on towards the Montage crowd, I'm afraid ...

**Comment by fpavogt** [comment]
For the record, the same underlying Montage issue with nan's also wreaks havoc with ``make_rgb_cube`` for which Montage is used without choice. As with the example illustrated above, creating an RGB image from images containing nan's will lead to a different number of pixels displayed in the resulting RGB frame.

**Comment by astrofrog** [comment]
@fpavogt - could you check if you still have this issue with the developer version of APLpy? We no longer use Montage, so I'd like to check if this still an issue.

**Comment by fpavogt** [comment]
developer version = current state of the Github repo, right ?

**Comment by fpavogt** [comment]
Alright, using the current aplpy Github master branch for testing purposes, the problem seems to be a bit different now. Namely, the nan regions "grow" when I set north = True. Before, it was the data regions that were growing. 

Here's an image to illustrate this. The nan regions are in dark blue. Everything else is valid data. All stripes are 1 pixel wide. If I set north=False, it all looks good:

![north_false](https://user-images.githubusercontent.com/13974615/37110770-96d5a7e6-221c-11e8-823c-497f6c8b35f6.png)

If I set north=True, then the nan regions grow by~1 row (at the expanse of areas with valid data)

![north_true](https://user-images.githubusercontent.com/13974615/37110771-96ff97b8-221c-11e8-8cb3-67fc347ea9ac.png)

So, as far as I can tell, there's also something funky going on with reproject (I installed v0.4) and nan.
